tools/sklearn_tools/numeric_clustering.xml

```diff
@@ -50,13 +50,20 @@ from scipy.io import mmread
 input_json_path = sys.argv[1]
 params = json.load(open(input_json_path, "r"))
 
-my_class = getattr(sklearn.cluster, params["algorithm_options"]["selected_algorithm"])
+selected_algorithm = params["algorithm_options"]["selected_algorithm"]
+
+my_class = getattr(sklearn.cluster, selected_algorithm)
 cluster_object = my_class()
 options = params["algorithm_options"]["options"]
+
 cluster_object.set_params(**options)
 
 #if $input_types.selected_input_type == "sparse":
-data_matrix = mmread(open("$infile", 'r'))
+if selected_algorithm in ["Birch","DBSCAN","KMeans","MiniBatchKMeans","SpectralClustering"]:
```

Comment:
You could put the `<conditional name="algorithm_options">` into the sparse conditional and remove/add algorithm that are only supported by sparse ... Not sure if this is easier for the user, but it would catch this before submitting the job.